api/db.py

```diff
@@ -42,6 +42,14 @@ async def find_by_id(self, model, obj_id):
         obj = await col.find_one({'_id': ObjectId(obj_id)})
         return model(**obj) if obj else None
 
+    async def find_by_attribute(self, model, query_list):
+        col = self._get_collection(model)
+        objs = []
+        data = col.find(query_list)
+        async for obj in data:
+            objs.append(model(**obj))
```

Comment:
This should be using `await` rather than `async`, as `async` is to declare functions that will be running asynchronous commands whereas `await` is when calling code that runs asynchronous commands.

Maybe the syntax could be simplified using list comprehension (I haven't tried to run this):
```python
data = col.find(attributes)
return list(await model(**obj) for obj in data)
```

Reply:
I used ```return await data.to_list(None)``` for it.

Reply:
OK, but that's not the same as this:
```python
        data = await col.find(attributes).to_list(None)
        return list(model(**obj) for obj in data)
```
The returned value from `data.to_list(None)` is a list of dictionaries, whereas the code above returns a list of `Node` objects.  The `Node` objects then get converted to JSON by `api.main` in the same way as the dictionaries so the data returned by the API looks the same, but it's not quite the same from an implementation point of view.  The `find_by_id()` method returns a `Node` object using `return model(**obj)` so I think it would be better to stay consistent with the `find_by_attributes()` method.  Other parts of the code may be using `api.db` and expect actual `Node` objects rather than dictionaries.

Reply:
Yes, makes sense.